meshapp/src/components/SignUp/inputs/CustomSelect.tsx

```diff
@@ -0,0 +1,91 @@
+import { Controller, useFormContext } from "react-hook-form";
+import { IFormInput } from "../SignUp";
+import {
+  Box,
+  FormControl,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Select,
+  Tooltip,
+} from "@mui/material";
+import ErrorIcon from "@mui/icons-material/Error";
+
+const menuStyle = { style: { maxHeight: 200 } };
+
+//takes in a field name, boolean for whether its required or not, label, id, list of options to display, a callback, and validator rules
+export default function CustomSelect(args: {
+  fieldName: "country" | "state" | "userType";
+  required?: boolean;
+  label?: string;
+  id?: string;
+  options?: Array<string>;
+  callback?: () => void;
+  validators?: { [key: string]: (value: string) => boolean | string };
+}) {
+  const { control, formState, trigger } = useFormContext<IFormInput>();
+  const errors = formState.errors;
+
+  return (
+    <Grid sx={{ position: "relative", alignItems: "center" }}>
+      <FormControl fullWidth margin="normal">
+        <InputLabel shrink={true}>{args.label}</InputLabel>
+        <Controller
+          name={args.fieldName}
+          control={control}
+          rules={{
+            required: args.required ? "This is required" : false,
+            validate: args.validators,
+          }}
+          render={({ field: { onChange, value } }) => (
+            <Select
+              displayEmpty
+              notched={true}
+              value={value}
+              onChange={(event) => {
+                onChange(event);
```

Comment:
When I run your branch myself, I get an error on these lines, specifically with onChange(event); so it might be worth checking if there are some dependency/package issues here.

Reply:
There appears to be no problems when React-Hook-Form is on version 7.44.2. Upon making the PR I merged the two differing package.json together, and I chose the higher version 7.45.1, which results in this error happening. Im not sure what to do other than changing the version back.

Reply:
actually nvm i fixed it

Reply:
Good find :) 🥇